.github/workflows/win_ci.yml

```diff
@@ -0,0 +1,302 @@
+name: WIN
+
+on:
+  push:
+    paths-ignore:
+      - 'CHANGES*'
+      - 'Doc/**'
+      - 'appveyor.yml'
+  pull_request:
+    branches: master
+    paths-ignore:
+      - 'CHANGES*'
+      - 'Doc/**'
+      - 'appveyor.yml'
+
+permissions:
+  contents: read
+
+jobs:
+  win_ci:
+    # When continue-on-error is true for an individual build,
+    # that build can fail (it'll show red),
+    # but it won't fail the overall tests
+    continue-on-error: ${{ matrix.continue-on-error || false }}
+
+    # https://github.com/actions/runner-images/blob/main/images/windows/Windows2022-Readme.md
+    runs-on: ${{ matrix.os || 'windows-2022' }}
+
+    # The name of the test follow the tested language
+    name: >
+      ${{ matrix.SWIGLANG }} ${{ matrix.VER }}
+      ${{ matrix.COMPILER || 'msvc' }} ${{ matrix.os }}
+      ${{ matrix.continue-on-error && '(can fail)' }}
+
+    strategy:
+      matrix:
+        include:
+        - SWIGLANG: csharp
+          INSTALL: 'true'
+        - SWIGLANG: csharp
+          os: 'windows-2019'
+          INSTALL: 'true'
+        - SWIGLANG: java
+          VER: 8
+        - SWIGLANG: java
+          VER: 11
+        - SWIGLANG: java
+          VER: 17
+       #  TODO JDK 21 fails on testing
+       #- SWIGLANG: java
+       #  VER: 21
+        - SWIGLANG: python
+          VER: '3.7'
+        - SWIGLANG: python
+          VER: '3.8'
+        - SWIGLANG: python
+        # VER: '3.9'
+        - SWIGLANG: python
+          VER: '3.10'
+        - SWIGLANG: python
+          VER: '3.11'
+        - SWIGLANG: python
+          VER: '3.12'
+        - SWIGLANG: python
+          INSTALL: 'true'
+          COMPILER: gcc
+      # Run all of them, as opposed to aborting when one fails
+      fail-fast: false
+
+    env:
+      SWIGJOBS: -j2
+      CFLAGS: '-O2'
+      CXXFLAGS: '-O2'
+      CCCL_OPTIONS: '--cccl-muffle /W3 /EHsc'
+      PCRE2_CCCL_LD: '-lpcre2-8-static --cccl-link /NODEFAULTLIB:MSVCRT'
+      CHECK_OPTIONS: 'CSHARPOPTIONS=-platform:x64'
+      SWIGLANG: ${{ matrix.SWIGLANG }}
+      COMPILER: ${{ matrix.COMPILER }}
+      INSTALL: ${{ matrix.INSTALL }}
+      VER: ${{ matrix.VER }}
+      OS: ${{ matrix.os }}
+
+# cl.exe:
+#  https://learn.microsoft.com//cpp/build/reference/compiler-options
+#  /EHc	 extern "C" defaults to nothrow.
+#  /EHs	 Enable C++ exception handling (no SEH exceptions).
+#  /W3   Warning level.
+#  /WX   Treat warnings as errors.
+#  https://learn.microsoft.com//cpp/build/reference/linker-options
+#  cl.exe pass linker:
+#  /VERBOSE:LIB        Outputs progress messages during the link process.
+#  /NODEFAULTLIB:lib   Ignore library 'lib'
+
+    steps:
```

Comment:
I'd like to see a 'Machine Info' step added, like in the Linux CI.

Reply:
I added 
"systeminfo | findstr /B /C:"OS Name" /B /C:"OS Version"
On AppVeyor, is it sufficient?

Reply:
Yup, but add it to a 'Machine Info' step like on Linux CI please.

Reply:
Sure